hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/RecordStreamSigningUtils.java

```diff
@@ -0,0 +1,287 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.cli.sign;
+
+import static com.hedera.node.app.hapi.utils.exports.recordstreaming.RecordStreamingUtils.readMaybeCompressedRecordStreamFile;
+import static com.hedera.services.stream.proto.SignatureType.SHA_384_WITH_RSA;
+import static com.swirlds.common.stream.LinkedObjectStreamUtilities.readFirstIntFromFile;
+import static com.swirlds.platform.util.FileSigningUtils.signData;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.UnsafeByteOperations;
+import com.hedera.node.app.hapi.utils.exports.FileCompressionUtils;
+import com.hedera.services.stream.proto.HashAlgorithm;
+import com.hedera.services.stream.proto.HashObject;
+import com.hedera.services.stream.proto.RecordStreamFile;
+import com.hedera.services.stream.proto.SignatureFile;
+import com.hedera.services.stream.proto.SignatureObject;
+import com.swirlds.common.crypto.DigestType;
+import com.swirlds.common.crypto.HashingOutputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SignatureException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * Utility class for signing record stream files
+ */
+public class RecordStreamSigningUtils {
+    /**
+     * Hidden constructor
+     */
+    private RecordStreamSigningUtils() {}
+
+    /**
+     * Supported stream version file
+     */
+    public static final int SUPPORTED_STREAM_FILE_VERSION = 6;
+
+    /**
+     * a messageDigest object for digesting entire stream file and generating entire record stream
+     * file hash
+     */
+    private static MessageDigest streamDigest;
+
+    /**
+     * a messageDigest object for digesting metaData in the stream file and generating metaData
+     * hash. Metadata contains: record stream version || HAPI proto version || startRunningHash ||
+     * endRunningHash || blockNumber, where || denotes concatenation
+     */
+    private static MessageDigest metadataStreamDigest;
```

Comment:
Aren't these two `private static` fields - a sign that this class is _not_ in fact a utility class with only static methods.   That it should be an _actual proper_ class you create an instance of, that holds its own state?  This is "global variable" territory, as written. 

`static` methods are for true utility functions, in fact, I'd go further and say they're only for true _standalone_ utility functions.  If functions have dependencies on calling other functions and using the state they create/modify - they shouldn't be static.  This should be a real class.  IMO.

When you make it a real class suddenly you can declare them `final` and initialize them in the constructor.  Then it's clear what's going on.

(Now I look further at the code it's not clear to me why you need two of them (they support a `reset` operation) or why they need to be preallocated.  Use only one - with reset - or just create them when you need them. (Encapsulate the use in a method so you can catch the useless `NoSuchAlgorithm` exception.)

Reply:
Check createSignatureFile(), you will see why you need to two of them. Since signRecordStreamFile() is static and called in RecordStreamSignCommand following event stream sign pattern from platform, all other methods become static. Non static works as well.